### DISCUSSION TITLE
Videogame laggy

### DISCUSSION THREAD

**Discussion by Gedankenn** [prasanthrangan/hyprdots#807] [Q&A] [answered]
There is frame flickering in videogame and in vlc media player. anyone know how to solve it please?

**Comment by zer0cha0s** [comment]
Hey could you please provide the Graphics card you have and also the driver version you are running.

**Comment by kRHYME7** [comment]
https://wiki.hyprland.org/Configuring/Variables/#opengl

**Comment by Gedankenn** [comment]
04:00.0 VGA compatible controller: NVIDIA Corporation GA104 [GeForce RTX 3060] (rev a1)
	Subsystem: Gigabyte Technology Co., Ltd GA104 [GeForce RTX 3060]
	Kernel driver in use: nvidia
	
Driver Version: 545.29.06    CUDA Version: 12.3 

Sorry for the delay, only could sit solve this now.

**Reply by kRHYME7** [reply]
![image](https://github.com/prasanthrangan/hyprdots/assets/53417443/784149fb-03db-472c-9f26-e624e182d4c0)

have you tried this ? 
https://wiki.hyprland.org/Configuring/Variables/#opengl

Some of this are already set  but maybe some can help  
https://wiki.hyprland.org/hyprland-wiki/pages/Nvidia/

**Reply by Gedankenn** [reply]
thanks, i will try those above. i know it is some wayland problem becouse when i play games with x11 i have no such flickerings. (but i fucking love this rice, and wayland, i dont want to be switching everytime i wanna play some game.) 

i will try this tonight i let you guys know if it fix the problem. really thanks for the support =D

And, so you guys know, i am using a VM, qemu with passthrough, i run unraid in my machine.

**Reply by kRHYME7** [reply]
>https://wiki.hyprland.org/Configuring/Variables/#opengl

oh no this is defaulted to true this is already activated. 


And I guess you've don't this already right? 
https://wiki.hyprland.org/Getting-Started/Master-Tutorial/#vm
```
VM?
If not using a VM, skip this step

In a VM, make sure you have 3D acceleration enabled in your virtio config (or virt-manager) otherwise Hyprland will not work.

You can also passthru a GPU to make it work.

Please bear in mind 3D accel in VMs may be pretty slow.
```

**Reply by Gedankenn** [reply]
I passthru the entire gpu to the vm, the nvme where all is instaled also. the cores i use for the vm are also isolated. in practice it should not matter. (i am thinking i might have some cpu cores problem, and i need to passthru more cores i am passing 4 cores 8 threads). amd ryzen 2700x

**Reply by kRHYME7** [reply]
Hmmm,  

I can only find this. 
http://wiki.hyprland.org/hyprland-wiki/pages/Nvidia/#fixing-random-flickering-nuclear-method
> Fixing random flickering, (nuclear method)
Do note though that this forces performance mode to be active, resulting in increased power-consumption (from 22W idle on a RTX 3070TI, to 74W).

>This may not even be needed for some users, only apply these ‘fixes’ if you in-fact do notice flickering artifacts from being idle for ~5 seconds.

>Make a new file at /etc/modprobe.d/nvidia.conf and paste this in:

>options nvidia NVreg_RegistryDwords="PowerMizerEnable=0x1; PerfLevelSrc=0x2222; PowerMizerLevel=0x3; PowerMizerDefault=0x3; PowerMizerDefaultAC=0x3"
Reboot your computer and it should be working.

>If it does not, try:

>lowering your monitors’ refresh rate, as this can stop the flickering altogether
installing the 535xx versions of the drivers, as later (545, 550) can cause flickering with XWayland
these are available for arch via [the AUR here](https://aur.archlinux.org/packages?O=0&K=535xx)
using the [Nouveau driver](https://wiki.archlinux.org/title/Nouveau) as mentioned above

**Reply by Gedankenn** [reply]
i tryed it, still flickering, even in vlc videos... i remmember i didnt had any flickerings with old nvidia driver versions, perhaps i will try it.

**Reply by zer0cha0s** [reply, marked as answer]
I had to downgrade to 535 and im still running that version now and its been fine. 
I did see there is a beta 550 version which is suppose to resolve some of the issues 545 created but dont quote me on that.

**Reply by Gedankenn** [reply]
Thanks, i will try giving more resources to the vm, and see how it plays, if nothing solves it i will downgrade. really thanks for the reply. =D

**Comment by pallaswept** [comment]
> I had to downgrade to 535 and im still running that version now and its been fine. I did see there is a beta 550 version which is suppose to resolve some of the issues 545 created but dont quote me on that.

Tried it myself and also read reports on nvidia forums which concur, the issues introduced with 545 are not resolved in the 550 beta.

Note that the Windows 545 drivers had similar issues and they also existed in the initial Windows 550 release and were fixed in the most recent Windows 550 series 'hotfix' driver.

Maybe when our linux driver comes out of beta, we might get the same fixes.

**Comment by Gedankenn** [comment]
just for curiosity, i changed my monitor to a 4k one, previous i was using a full hd monitor with 140 frame rate
Now, the lag has vanished, my gpu is always at 99%. 

my theory is that the linux was trying to put monitor at 140fps while the game could only handle 60.  if a come across another full hd monitor i will try to force it in linux to be 60 fps.

**Reply by pallaswept** [reply]
I also observe these graphical flaws outside of games - for example, rendering videos on the desktop: occasionally a youtube video or a video played from VLC or MPV will flash a single frame which is from quite some time in the past. 

Using the frametime monitor built into the DE, I can see that render times for those frames - which are usually in the region of 2-3ms per frame - are well above 16.6` ms (60FPS) and out of curiosity regarding my multi-framerate desktop configuration, I did try forcing my monitors to 60Hz (usually, one is full HD at 60 and one is QHD at 165) and out of desperation, even tried 30Hz... sadly, it didn't do the trick, either in games, benchmarks, or on the desktop. Furthermore, I tried disabling each of the displays and running the single display at 60Hz, as well as disabling vsync. No change.

I also limited the buffer depth to a single frame, since it was showing me old frames, I figured I'd remove it's ability to do so by not retaining said old frames in the buffer. Again, no luck. Even without those old frames being buffered, as in, they shouldn't even exist in memory any more, they were still displayed in these momentary glitches. Obviously there's an issue with the drivers' handling of the frame buffer allocation/reading.

Regarding utilisation, I also observe this behaviour with the card locked to the P0 pstate (the 'nuclear option' suggested above), and when using high (99-100%) or low (~30%) utilisation.

You might have just gotten lucky, or maybe if you test a little longer you'll see the glitches come back. As I mentioned above, they are a known issue in these drivers, so you should expect to see them. I hope you stay lucky and can continue to enjoy the games without any problems, though!

**Reply by Gedankenn** [reply]
Wow, you tested it thoroughly. I hope my problem has gone away. I am anxious to upgrade my setup. I found this kit server on AliExpress. I can only dream for now. Hehehe. [kit server aliexpress](https://pt.aliexpress.com/item/1005006470573275.html?spm=a2g0o.productlist.main.35.afbe5a60rHsLzL&algo_pvid=4f519f7b-70cf-4f56-918a-04b2101969c4&algo_exp_id=4f519f7b-70cf-4f56-918a-04b2101969c4-17&pdp_npi=4%40dis%21BRL%214079.38%212488.44%21%21%21778.24%21474.73%21%402101fb0f17075357189011973e4081%2112000037316079333%21sea%21BR%210%21AB&curPageLogUid=HmDYsZAGuwx0&utparam-url=scene%3Asearch%7Cquery_from%3A)

**Reply by pallaswept** [reply]
Dual socket x99, NICE! Probably don't *need* to upgrade that any time real soon. My single-socket X99 machine still going like a trooper. With all that CPU juice, that thing will still kick butt as a server for a long time to come.

Good luck with the games anyway bruh! It's the weekend, time to "thoroughly stress test it" (read: now you have a wicked good excuse to play games all weekend lol(
